sycl/include/CL/sycl/detail/program_manager/program_manager.hpp

```diff
@@ -42,7 +42,9 @@ using DeviceImage = pi_device_binary_struct;
 struct ImageDeleter;
 
 enum DeviceLibExt {
-  cl_intel_devicelib_assert = 0
+  cl_intel_devicelib_assert = 0,
+  cl_intel_devicelib_math,
+  cl_intel_devicelib_complex
```

Comment:
What is the logic behind separating device library?
Is there any document describing where should new device function go?

Reply:
It seems that we don't have such documentation.  In current device lib design, if device provides native functions to override DPC++ compiler's fallback implementation, our fallback device lib won't be loaded by sycl runtime. Suppose we put everything in just one device lib including complex, cmath...., if device only provide native support for cmath functions but nothing for complex and assert, the fallback devicelib won't be loaded and the complex/assert will not work.

Reply:
Please, document the design.

Reply:
Already documented in C-CXX-StandardLibrary.rst.